src/revChatGPT/ChatGPT.py

```diff
@@ -125,57 +171,264 @@ def reset_chat(self) -> None:
-                            # remove the semicolon and 'cf_clearance=' from the string
-                            raw_cookie = cookie.group(0)
-                            self.cf_clearance = raw_cookie[13:-1]
-                            self.cookie_found = True
-
-        def detect_user_agent(message):
-            if 'params' in message:
-                if 'headers' in message['params']:
-                    if 'user-agent' in message['params']['headers']:
-                        # Use regex to get the cookie for cf_clearance=*;
-                        self.user_agent = message['params']['headers']['user-agent']
-                        self.agent_found = True
         options = uc.ChromeOptions()
+        options.add_argument('--start_maximized')
         options.add_argument("--disable-extensions")
         options.add_argument('--disable-application-cache')
         options.add_argument('--disable-gpu')
         options.add_argument("--no-sandbox")
         options.add_argument("--disable-setuid-sandbox")
         options.add_argument("--disable-dev-shm-usage")
+        print("Spawning browser...")
         driver = uc.Chrome(enable_cdp_events=True, options=options)
+        print("Browser spawned.")
         driver.add_cdp_listener(
-            "Network.responseReceivedExtraInfo", lambda msg: detect_cookies(msg))
+            "Network.responseReceivedExtraInfo", lambda msg: self.detect_cookies(msg))
         driver.add_cdp_listener(
-            "Network.requestWillBeSentExtraInfo", lambda msg: detect_user_agent(msg))
+            "Network.requestWillBeSentExtraInfo", lambda msg: self.detect_user_agent(msg))
         driver.get("https://chat.openai.com/chat")
-        while not self.agent_found or not self.cookie_found:
+        while not self.agent_found or not self.cf_cookie_found:
             sleep(5)
         driver.quit()
         del driver
         self.refresh_headers(cf_clearance=self.cf_clearance,
                              user_agent=self.user_agent)
+    def detect_cookies(self, message):
+        if 'params' in message:
+            if 'headers' in message['params']:
+                if 'set-cookie' in message['params']['headers']:
+                    # Use regex to get the cookie for cf_clearance=*;
+                    cf_clearance_cookie = re.search(
+                        "cf_clearance=.*?;", message['params']['headers']['set-cookie'])
+                    session_cookie = re.search(
+                        "__Secure-next-auth.session-token=.*?;", message['params']['headers']['set-cookie'])
+                    if cf_clearance_cookie and not self.cf_cookie_found:
+                        print("Found Cloudflare Cookie!")
+                        # remove the semicolon and 'cf_clearance=' from the string
+                        raw_cf_cookie = cf_clearance_cookie.group(0)
+                        self.cf_clearance = raw_cf_cookie.split("=")[1][:-1]
+                        if self.verbose:
```

Comment:
Rather than having a ton of self.verbose, I think it would be better to have a Logger class that prevents the repetitive if statements. I will do that.

Reply:
sounds good